server/server.py

```diff
@@ -78,6 +49,42 @@ def hello():
     return jsonify(top.current_user)
 
 
+@app.route("/record/<record_id>", methods=('GET', 'PUT'))
+@cross_origin(headers=["Content-Type", "Authorization"])
+@cross_origin(headers=["Access-Control-Allow-Origin", "*"])
+def record(record_id):
+    jwt = request.headers.get("Authorization", b'')
+    filename = os.path.join(HERE, "records", "{record_id}.json".format(
+        record_id=os.path.basename(record_id)))
+    new = True
+    if os.path.exists(filename):
+        new = False
+        with open(filename, 'r') as f:
+            record = json.load(f)
+            author = record["author"]
+    if request.method == "GET":
+        # READ
+        allowed(server=IAM_SERVER, audience=API_AUDIENCE,
+                resource="record", action="read", jwt=jwt, context={"author": author})
+        return jsonify(record['body'])
+
+    elif request.method == "PUT":
+        if new:
+            payload = allowed(server=IAM_SERVER, audience=API_AUDIENCE,
+                              resource="record", action="create", jwt=jwt)
+        else:
+            payload = allowed(server=IAM_SERVER, audience=API_AUDIENCE,
+                              resource="record", action="update", jwt=jwt,
+                              context={"author": author})
+
+        with open(filename, 'w') as f:
+            body = {'body': request.get_json(), 'author': payload["principals"][1]}
```

Comment:
use intermediary variable `email` ?

Reply:
actually, I'm not sure I guarantee that email is the second one. You could iterate on principals, and just simplify and use the first one...